quickwit/quickwit-doc-mapper/src/default_doc_mapper/mapping_tree.rs

```diff
@@ -377,6 +584,37 @@ impl MappingNode {
         }
     }
 
+    /// Finds the field mapping type for a given field path in the mapping tree.
```

Comment:
I know this is not part of this PR, but can we simplify `573-577` ? It seems like the look up in branches order makes no sense.